### PR TITLE
Refactoring Changes

### PR DESCRIPTION
**Pull request by msahilan07**
## What

Code Improvement 
Adding Tests

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/main/java/gov/uk/courtdata/unlink/validator/UserIdValidator.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
     @Override
     public Optional<Void> validate(String userId) throws ValidationException {
 
-        if (!userId.isEmpty() && userId.length()>0) {
+        if (userId != null) {
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/unlink/validator/UserIdValidator.java:19]: Don't we need to check if the user-id is an empty string or not?

Comment (msahilan07) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/unlink/validator/UserIdValidator.java]: Good Point. Fixed it now!

File: maat-court-data-api/src/test/java/gov/uk/courtdata/unlink/validator/UserIdValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private UserIdValidator userIdValidator;
+
+
+
+    @Test
+    public void testUserIdValidator_whenUserIsNullThrowsException() {
+
+        thrown.expect(ValidationException.class);
+        thrown.expectMessage("User id is missing.");
+        userIdValidator.validate(null);
```
Comment (atharmajeed) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/unlink/validator/UserIdValidatorTest.java:26]: we might need to add another test for empty string.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/model/LaaTransactionLogging.java [hunk 3]
```diff
@@ -13,9 +13,9 @@
 @Value
 public class LaaTransactionLogging {
 
-    private UUID laaTransactionId;
-    private Integer maatId;
-    private String caseUrn;
+    UUID laaTransactionId;
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/model/LaaTransactionLogging.java:16]: Just being pedantic here, we should restrict the access fields and they should only be accessible within a class, the default value is package-private unless there is a reason I'm not aware of. I'm fine either way.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/unlink/controller/UnLinkController.java [hunk 4]
```diff
@@ -2,33 +2,32 @@
...
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/unlink")
 @Slf4j
+@RequiredArgsConstructor
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/unlink/controller/UnLinkController.java:14]: This is good.

File: maat-court-data-api/src/test/java/gov/uk/courtdata/unlink/validator/UserIdValidatorTest.java (hunk 2 above)
Comment (msahilan07) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/unlink/validator/UserIdValidatorTest.java:26]: Added, also added to reason ID validator.
